lib/ConditionBuilder.js

```diff
@@ -1,69 +1,89 @@
 var assert = require('assert')
 var typ = require('typ')
 var typeUtil = require('./typeUtil')
-var ConditionJunction = require('./ConditionJunction')
+var util = require('util')
 
 /**
- * @param {Object=} opt_conditions
+ * @param {Object=} opt_exprs
  * @constructor
  */
-function ConditionBuilder(opt_conditions) {
-  this._conditions = opt_conditions || {}
-  this._uniqueName = ''
+function ConditionBuilder(opt_exprs) {
+  this._exprs = opt_exprs || []
 }
 
 /**
- * @param {Array.<ConditionBuilder|ConditionJunction>} conditions
- * @return {ConditionJunction}
+ * Creates a new Conditional expression with an operator (AND, EQ, etc)
+ */
```

Comment:
good point. i defined some better types for these that will hopefully make it clearer and more constrained.